lib/src/__tests__/DatePickerTestingLib.test.tsx

```diff
@@ -21,4 +22,56 @@ describe('<DatePicker />', () => {
 
     expect(screen.getByRole('button', { name: '2010' })).toBeDisabled();
   });
+
+  it("Doesn't close picker on selection in Mobile mode", () => {
+    render(
+      <MobileDatePicker
+        value={utilsToUse.date('2018-01-01T00:00:00.000Z')}
+        onChange={jest.fn()}
+        renderInput={props => <TextField {...props} />}
+      />
+    );
+
+    fireEvent.click(screen.getByRole('textbox'));
+    fireEvent.click(screen.getByLabelText('Jan 2, 2018'));
+
+    expect(screen.queryByRole('dialog')).toBeInTheDocument();
+  });
+
+  it('Closes picker on selection in Desktop mode', async () => {
+    render(
+      <DesktopDatePicker
+        TransitionComponent={FakeTransitionComponent}
+        value={utilsToUse.date('2018-01-01T00:00:00.000Z')}
+        onChange={jest.fn()}
+        renderInput={props => <TextField {...props} />}
+      />
+    );
+
+    fireEvent.click(screen.getByLabelText('Choose date, selected date is Jan 1, 2018'));
+
+    await waitFor(() => screen.getByRole('dialog'));
```

Comment:
I think that we should look into why it's necessary:

```suggestion
    screen.getByRole('dialog');
```

Reply:
Yes, it is an issue with popper native `onCreate` function that works through timeout and need to update the state. (in order to allow focusing elements and prevent scroll)